File: P120875.py
# 쓰레기 문제

def get_res(a, b, c, d):
    
    if (b[1] - a[1]) * (d[0] - c[0]) == (d[1] - c[1]) * (b[0] - a[0]):
        return 1
    
    return 0

def solution(dots):
    
    if get_res(dots[0], dots[1], dots[2], dots[3]):
        return 1
    
    if get_res(dots[0], dots[2], dots[1], dots[3]):
        return 1
    
    if get_res(dots[0], dots[3], dots[1], dots[2]):
        return 1
    
    return 0

File: P120875.java
public class P120875 {
    public int solution(int[][] dots) {
        if (isParel(dots[0], dots[1], dots[2], dots[3]) || isParel(dots[0], dots[2], dots[1], dots[3]) || isParel(dots[0], dots[3], dots[1], dots[2])) return 1;
        return 0;
    }
    public boolean isParel(int[] x, int[] y, int[] z, int[] w) {
        if ((x[0] - y[0]) * (z[1] - w[1]) == (x[1] - y[1]) * (z[0] - w[0])) return true;
        return false;
    }
}
